staging/src/k8s.io/apiserver/pkg/endpoints/handlers/create.go

```diff
@@ -94,6 +96,14 @@ func createHandler(r rest.NamedCreater, scope RequestScope, typer runtime.Object
 		audit.LogRequestObject(ae, obj, scope.Resource, scope.Subresource, scope.Serializer)
 
 		userInfo, _ := request.UserFrom(ctx)
+		// On create, get name from new object if unset
+		if len(name) == 0 {
+			if metadata, err := meta.Accessor(obj); err == nil {
+				name = metadata.GetName()
+			} else {
+				glog.Warningf("object %#v has no meta: %v", obj, err)
```

Comment:
A failure here changes the invariant to the admission.  You need to fail, right?